straxen/analyses/waveform_plot.py

```diff
@@ -211,15 +247,113 @@ def plot_peak(p, t0=None, **kwargs):
 def time_and_samples(p, t0=None):
     """Return (x, y) numpy arrays for plotting the waveform data in p
     using 'steps-pre'.
-
     Where x is the time since t0 in seconds (or another time_scale),
       and y is intensity in PE / ns.
     :param p: Peak or other similar strax data type
     :param t0: Zero of time in ns since unix epoch
     """
-    n = p['length']
+    n = int(p['length'])
     if t0 is None:
         t0 = p['time']
     x = ((p['time'] - t0) + np.arange(n + 1) * p['dt']) / int(1e9)
     y = p['data'][:n] / p['dt']
     return x, np.concatenate([[y[0]], y])
+
+
+def plot_wf(st: strax.Context,
+            containers,
+            runs, plot_log=True, plot_extension=5_000, hit_pattern=True,
+            timestamp=True, time_fmt="%d-%b-%Y (%H:%M:%S)",
+            **kwargs):
+    """
+    Combined waveform plot
+    :param st: strax.Context
+    :param containers: peaks/records/events where from we want to plot all the peaks that
+        are within it's time range +- the plot_extension
+    :param runs: runs (data frame or single run_number)
+    :param plot_log: Plot the y-scale of the wf in log-space
+    :param plot_extension: include this much time arround the contaiers (can be scalar or
+        list of (left_extension, right_extension)
+    :param hit_pattern: include the hitpattern in the wf
+    :param timestamp: print the timestamp to the plot
+    :param time_fmt: format fo the timestamp (datetime.strftime format)
+    :param kwargs: kwargs for plot_peaks
+    """
+    p = containers  # usually peaks
+
+    if isinstance(runs, str):
+        print(f'Assuming {runs} is a run id')
+        runs = pd.DataFrame({'name': [runs]})
+
+    def get_run_start(peak):
```

Comment:
I think I do not fully catch the idea behind this function. Can you describe the users use case for this plot function and how to call it? To me `containers` implies somehow I can specify an iterable of things, while when reading the code I get the impression it is just for a single event peak etc.

I understand it in the following way: The user works with peaks/events from several runs in his/her analysis and spots maybe some odd behavior and would like to plot the waveform for the corresponding peaks. Hence he/she puts the corresponding list of e.g. `events` into the argument `containers` and specifies in `runs` which `run_ids` were used. Since the user does not know to which run the events belongs to, you call this function `get_run_start`? `get_run_start` searches now for the last run_id which end_time is smaller than the last containers (events) start time.   

A bit further down you plot then all peaks which are within this run_id and the corresponding time of the containers. But what happens if the containers (events) are distributed over more than one run? I have the impression you would chop all peaks which do not belong to the very last run?

Reply:
matbe I should indeed update the docstring. We are assuming that you are plotting peaks that are close to one another (why would you want to have peaks over multiple runs anyway). They are plotted in a single figure, I think that is the key information you are missing.

We need to have the run_id, so you either specify it or you just give a list that the function itself with figure out where the group of peaks belongs to (assuming all are in the same).

Reply:
Okay thanks for clarification. Now it makes indeed more sense. I was in the beginning more thinking about some sort of interactive plot where you can move your view back and forth in time. Then it make sense.

Reply:
Ok, just simplified a bit